src/pages/en/guides/content-collections.md

```diff
@@ -0,0 +1,375 @@
+
+const announcementPost = await getEntry('announcements', 'welcome.md');
+const { Content } = await renderEntry(announcementPost);
+---
+
+<h1>{announcementPost.data.title}</h1>
+<Content />
+```
+
+### Access content headings
+
+Astro [generates a list of headings](/en/guides/markdown-content/#exported-properties) for Markdown and MDX documents. You can access this list using the `headings` property from `renderEntry`:
+
+```astro "{ headings }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { headings } = await renderEntry(post);
+  const h1 = headings.find(h => h.depth === 1);
+  return <p>{h1}</p>
+})}
+```
+
+### Access injected frontmatter
+
+Astro allows you to [inject frontmatter using remark or rehype plugins.](/en/guides/markdown-content/#example-injecting-frontmatter) You can access these values using the `injectedFrontmatter` property from `renderEntry`:
+
+```astro "{ injectedFrontmatter }"
+---
+import { getCollection, renderEntry } from 'astro:content';
+const blogPosts = await getCollection('blog');
+---
+
+{blogPosts.map(async (post) => {
+  const { injectedFrontmatter } = await renderEntry(post);
+  return <p>{post.data.title} — {injectedFrontmatter.readingTime}</p>
+})}
+```
+
+Assuming `readingTime` was injected ([see our reading time example](/en/guides/markdown-content/#example-calculate-reading-time)), it will be available on the `injectedFrontmatter` object.
+
+<details>
+<summary>**🙋 Why don't `getCollection` and `getEntry` contain these values?**</summary>
+
+The remark and rehype pipelines are only run when your content is **rendered.** This lets `renderEntry` access anything generated by these plugins like injected frontmatter. To stay performant, `getCollection` and `getEntry` do not have this capability.
+
+</details>
+
+## Generating pages from content collections
+
+You can create pages from your content collections using [dynamic routes](/en/core-concepts/routing/#dynamic-routes).
+
+Say you have a `src/content/blog/` collection, and you want to map these entries to `/posts/*` URLs. You can create a `posts/[...slug].astro` route using `getStaticPaths` like so:
+
+```astro "{ slug: entry.slug }"
+---
+// src/pages/posts/[...slug].astro
+import { getCollection } from 'astro:content';
+export async function getStaticPaths() {
+  const blog = await getCollection('blog');
+  // Map collection entries to pages
+  return blog.map(entry => ({
+    // `entry.slug` is the filename of each blog post with `.md` removed
+    params: { slug: entry.slug },
+  }));
+}
+---
+```
+
+This will generate routes for every entry in our collection, mapping each entry’s slug to a URL. For example, an entry at `src/content/blog/hello-world.md` will have a slug of `hello-world`. Because this dynamic route is in `src/pages/posts/`, the final URL will be `/posts/hello-world`.
```

Comment:
```suggestion
This will generate routes for every entry in our collection, mapping each entry's slug to a URL. For example, an entry at `src/content/blog/hello-world.md` will have a slug of `hello-world`. Because this dynamic route is in `src/pages/posts/`, the final URL will be `/posts/hello-world`.
```